core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -113,48 +115,21 @@ abstract class TaskContext extends Serializable {
    *
    * Exceptions thrown by the listener will result in failure of the task.
    */
+  @DeveloperApi
```

Comment:
I confirmed that the TaskCompletionListener and TaskFailureListener classes themselves have been marked as `DeveloperApi` since ~2014/2015 👍